firmware/library/L1_Drivers/onboardled.hpp

```diff
@@ -18,3 +20,63 @@ class OnBoardLedInterface {
            virtual void LedSetAll(uint8_t value)                     = 0;
            virtual uint8_t LedGetValues(void)                        = 0;
 };
+
+class OnBoardLed: public OnBoardLedInterface
+{
+ public:
+  // Initialization involes setting the appropriate pins to be GPIO, and
+  // then outputs. Once they are configured as such, they can be maniputlated
+  // as expected.
+  void Initialize(void) override
+    {
+    // This block will make sure all the GPIOs that were initialized in the
+    // last block will be set as outputs.
+    Led0.SetAsOutput();
+    // Led1.SetAsOutput();
+    // Led2.SetAsOutput();
+    // Led3.SetAsOutput();
+    // This block turns off all the LEDs by default, to save power.
+    led_states = 0xFF;
+    LedSetAll(led_states);
+  }
+
+  void LedOn(uint8_t led_number) override
```

Comment:
You can remove the Prefix of `LED` from all of the methods. Think about it this way, the developer will write some code like the following:

    OnboardLeds leds;
    leds.LedOn(1);

Having to write Led twice is unnecessary. The user can easily understand what it is that is happening if you omit it.

    OnboardLeds leds;
    leds.On(1);

Prefer to keep the method names full, but concise.

Reply:
I will change it as so. I figured a more full name would work out better.